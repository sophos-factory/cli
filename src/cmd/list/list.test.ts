/* eslint-disable @typescript-eslint/no-non-null-assertion */

import { execute } from '../../../tests/helpers/execute';
import { loadFixtures } from '../../../tests/fixtures';

beforeAll(async () => {
  return await loadFixtures();
});

describe('factoryctl list', () => {
  test('throws on missing subcommand', async () => {
    await expect(
      execute(['list'], { token: process.env.FACTORY_STATIC_AUTH_TOKEN! })
    ).rejects.toMatchSnapshot();
  });

  describe('credentials', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['list', 'credentials'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('jobs', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['list', 'jobs'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline-revisions', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['list', 'pipeline-revisions'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipelines', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['list', 'pipelines'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runner managers', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['list', 'runner-managers'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runs', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['list', 'runs'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });
});
