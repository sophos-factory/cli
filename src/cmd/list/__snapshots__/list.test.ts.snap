// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factoryctl list credentials throws on missing arguments 1`] = `
"factoryctl credentials [options]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JsonPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --offset  [number]
      --project-id    ID of project containing the credentials  [string] [required]
      --type  [string] [choices: "api_token", "aws_access_key", "azure_service_principal", "bearer_token", "generic", "ssh_key", "username_password", "vault_app_role", "google_service_account"]
      --sort          Sort order  [string] [choices: "created_asc", "created_desc"]
      --limit  [number]

Missing required argument: project-id
"
`;

exports[`factoryctl list jobs throws on missing arguments 1`] = `
"factoryctl jobs [options]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JsonPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --offset  [number]
      --project-id    ID of the project containing the jobs  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl list pipeline-revisions throws on missing arguments 1`] = `
"factoryctl pipeline-revisions [options]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JsonPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --offset  [number]
      --project-id    ID of the project containing the pipeline revision  [string] [required]
      --pipeline-id   ID of the pipeline containing the pipeline revision  [string] [required]
      --limit  [number]

Missing required arguments: project-id, pipeline-id
"
`;

exports[`factoryctl list pipelines throws on missing arguments 1`] = `
"factoryctl list pipelines

List pipelines for a project

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JsonPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --offset  [number]
      --project-id    ID of the project containing the pipelines  [string] [required]
      --sort          Sort order  [choices: "modified_asc", "modified_desc", "name_asc", "name_desc"]
      --limit  [number]

Missing required argument: project-id
"
`;

exports[`factoryctl list runner pools throws on missing arguments 1`] = `
"factoryctl list runner-pools

List runner pools for an organization

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JsonPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --offset  [number]
      --organization-id  ID of the organization containing the runner pools  [string]

Missing required argument: organization-id
"
`;

exports[`factoryctl list runners throws on missing arguments 1`] = `
"factoryctl list runners

List runners for an organization or project

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JsonPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --offset  [number]
      --organization-id  ID of the organization containing the runners  [string]
      --project-id       ID of the project containing the runners  [string]

Missing required argument: organization-id or project-id
"
`;

exports[`factoryctl list runs throws on missing arguments 1`] = `
"factoryctl runs [options]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JsonPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --offset  [number]
      --project-id    ID of the organization containing the runs  [string] [required]
      --pipeline-id   ID of a pipeline used to filter the runs  [string]
      --job-id        ID of a job used to filter the runs  [string]
      --sort          Sort order  [string] [choices: "created_asc", "created_desc"]
      --limit  [number]

Missing required argument: project-id
"
`;

exports[`factoryctl list throws on missing subcommand 1`] = `
"Usage: factoryctl list <command> [options]

Commands:
  factoryctl list credentials         List credentials from a project
  factoryctl list jobs                List jobs for a project
  factoryctl list organizations       List organizations to which authenticated user belongs
  factoryctl list pipelines           List pipelines for a project
  factoryctl list pipeline-revisions  List pipeline revisions for a pipeline
  factoryctl list projects            List projects from an organization
  factoryctl list runner-pools        List runner pools for an organization
  factoryctl list runners             List runners for an organization or project
  factoryctl list runs                List runs for a project

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JsonPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --offset  [number]

Command must be specified
"
`;
